### ISSUE TITLE
[Bug]: Deep link event is not working when app is closed and is opening through deep link alert window 

### ISSUE THREAD

**Issue by harsh-simform** [electron/electron#39462]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

21.4.1

### What operating system are you using?

Windows

### Operating System Version

Windows 11 22H2

### What arch are you using?

x64

### Expected Behavior

The app should listen to `open-url` or `second-instance` events for the incoming link from the browser. 

### Actual Behavior

`open-url` or `second-instance` event is not getting listened while app is opening.

### Additional Information

basically, it is only working when the app is running in the background.

**Comment by harsh-simform** [comment]
I have found a solution for this type of scenario where an app is closed, but it is getting opened by the deep link.
here, the listener `second-instance` is not getting fired but we can get a deep link from the mainWindow creation function's command line.

like this,

```typescript
   const commandLine = process.argv.slice(1);
    log.info('commandLine in mainWindow creation: ', commandLine);
    const url = commandLine[0];
```
